Initial microservice structure is finished

endlesslydivided opened the pull request.

endlesslydivided committed:
lerna configuration started and user_microservice finished

endlesslydivided committed:
user_microservice, material_microservice and lexic_microservice controllers, services, repositories and model schemas are created and base features are finished. Lerna monorepository configuration is created. Some errors are fixed in proto files. Linter and prettier configurations are created in each microservice project.

endlesslydivided committed:
Docker-compose config is created. API gateway started. User-stats microservice is edited

endlesslydivided committed:
User_stats_microservice<->Lexic_microservice and User_stats_microservice<->Material_microservice communitcation is established. User_stats_microservices debugged

IlyaRazuvaev commented on packages/user_microservice/src/auth/repository/user.repository.ts:
```diff
@@ -0,0 +1,144 @@
+  @InjectRepository(User)
+  private readonly userRepository: Repository<User>;
+
+  @InjectRepository(UserContacts)
```
Какой-то странный способ, почему не через конструктор

endlesslydivided committed:
API-gateway user-stats, lexic and material args,inputs, resolvers and models are created. API-gateway modules are configured.

endlesslydivided committed:
Auth enabled with guard and auth module completed with resolvers,models, args and inputs

endlesslydivided committed:
API-gateway debugged. GraphQL shcema is edited

endlesslydivided committed:
Create README.md

endlesslydivided committed:
Docker-compose config edited and fixed. Docker deployment is correct

endlesslydivided committed:
Kubernetes configMap, deployment and service yaml-configurations are finished and tested

endlesslydivided committed:
Client is created. Sign in and sign up forms are created.

endlesslydivided committed:
Client is created. Sign in and sign up forms are created

endlesslydivided committed:
Merge branch 'develop' of https://github.com/endlesslydivided/langua_app_microservices into develop

endlesslydivided committed:
User entity not null constraints are deleted and forms with types are edited for signUp

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/guard/auth.guard.ts:
```diff
@@ -0,0 +1,46 @@
+
+  public async canActivate(ctx: ExecutionContext): Promise<boolean> | never {
+    const req: Request = ctx.switchToHttp().getNext().req;
+    const authorization: string = req.headers['authorization'];
```
не делай так, пиши свои обертки над типами и extends от базовых

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/guard/auth.guard.ts:
```diff
@@ -0,0 +1,46 @@
+    const authorization: string = req.headers['authorization'];
+
+    if (!authorization) {
+      throw new UnauthorizedException();
```
всегда указывай почему ошибка произошла

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/guard/auth.guard.ts:
```diff
@@ -0,0 +1,46 @@
+
+    const token: string = bearer[1];
+
+    const { status, userId }: auth.ValidateResponse = await this.service.validate(token);
```
Все, что типы, пиши с большой буквы

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/guard/auth.guard.ts:
```diff
@@ -0,0 +1,46 @@
+
+    const { status, userId }: auth.ValidateResponse = await this.service.validate(token);
+
+    req['user'] = userId;
```
выше

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/middleware/refresh.middleware.ts:
```diff
@@ -0,0 +1,33 @@
+    {
+        try 
+        {
+            const cookie =req.header('cookie');
```
header?

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/resolver/auth.resolver.ts:
```diff
@@ -0,0 +1,40 @@
+
+    if(result.status !== HttpStatus.OK && result.status !== HttpStatus.CREATED)
+    {
+      throw new BadRequestException(result.error)
```
не нравиться, ошибки внутри других сервисов всегда badrequest. Либо сделай middleware и навесь на конец обработки реквеста и там кидай new HTTPException(result.status, err) (примерно), либо поищу нормальную либу

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/resolver/auth.resolver.ts:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    return {
+      accessToken: result.token
```
refresh?

IlyaRazuvaev commented on packages/langua_API_gateway/src/auth/service/user.service.ts:
```diff
@@ -0,0 +1,26 @@
+  }
+
+  public async findUserById(data: auth.FindUserByIdRequest): Promise<auth.FindUserByIdResponse> {
+    return firstValueFrom(this.authServiceClient.findUserById(data));
```
назвал сервис user_microservice, так его и называй

IlyaRazuvaev commented on packages/langua_API_gateway/src/lexic/service/word.service.ts:
```diff
@@ -0,0 +1,58 @@
+export class WordService {
+  private lexicServiceClient: lexic.LexicService;
+
+  @Inject(LEXIC_SERVICE_NAME)
```
почему property-based, а не конструктор?

IlyaRazuvaev commented on packages/langua_API_gateway/src/material/resolver/material.resolver.ts:
```diff
@@ -0,0 +1,85 @@
+        limit,
+      },
+    };
+    const result = await this.materialService.findManyMaterialsByCreatorId(data);
```
const { status, data } приятнее

IlyaRazuvaev commented on packages/langua_API_gateway/src/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+
+RUN npm install
+
+RUN npm ci
```
выбери install или ci

IlyaRazuvaev commented on packages/langua_API_gateway/src/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+FROM node:20-alpine
```
зачем внутри src

IlyaRazuvaev commented on packages/user_microservice/src/auth/repository/user.repository.ts:
```diff
@@ -0,0 +1,144 @@
+@Injectable()
+export class UserRepository {
+  @InjectRepository(User)
+  private readonly userRepository: Repository<User>;
```
https://docs.nestjs.com/providers

IlyaRazuvaev commented on packages/user_microservice/src/auth/auth.pb.ts:
```diff
@@ -0,0 +1,116 @@
+    export interface FindManyUsersRequest {
+        pageFilters?: auth.PageFilters;
+    }
+    export interface FindManyUsersResponse {
```
ErrorInterface | ResponseInterface

IlyaRazuvaev commented on packages/user_microservice/src/share/filter/http-exception.filter.ts:
```diff
@@ -0,0 +1,33 @@
+import { Request, Response } from 'express';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
```
это работает с grpc?

IlyaRazuvaev commented on packages/user-stats_microservice/src/user-stats.pb.ts:
```diff
@@ -0,0 +1,91 @@
+export const USER_STATS_PACKAGE_NAME = 'userStats'
+export const USER_STATS_SERVICE_NAME = 'UserStatsService'
+
+export namespace userStats {
```
в чем смысл использования namespace? 
Приложение маленькое, пересечения имен не будет

IlyaRazuvaev commented on packages/material_microservice/src/material/service/material.service.ts:
```diff
@@ -0,0 +1,153 @@
+  public async findManyMaterialsByVocabularyId(
+    dto: FindManyMaterialsByVocabularyIdRequestDto,
+  ): Promise<material.FindManyMaterialsByVocabularyIdResponse> {
+    const data = await this.materialRepository.findManyAndCountByVocabularyId(
```
кортежи удобно использовать через [rows, count]

IlyaRazuvaev commented on packages/lexic_microservice/src/vocabulary/repository/vocabulary.repository.ts:
```diff
@@ -0,0 +1,44 @@
+      )
+      .exec();
+
+    const count = await this.vocabularyModel.count(clause).exec();
```
больше нагрузка на бд, почему не использовать vcabulares.length

endlesslydivided committed:
Refresh token support is added

endlesslydivided committed:
Client authorization is fixed. Loader is added. Pages metadata provision fixed. Some api call-functions written.

endlesslydivided committed:
Providers injections are edited. Microservices filters are edited. Some other stuff is refactored

endlesslydivided committed:
1) Client authorization flow is finished
2) Exceptions tree is created on client side
3) Hook for paginated fetch is created on clien side
4) Frontend project structure is modified
5) Some errors and typos are fixed in api_gateway

endlesslydivided committed:
Theme on client is configured. LexicCategoriesList is created

endlesslydivided committed:
Some refactoring is done. Finish registration form is finished.

endlesslydivided committed:
1.Frontend project structure is fulle edited.
2.New frontend layout for authorized pages are created.
3.API calls for client lexicCategories and word fetching are created.
4.LexicCategories lists and creation form are created.
5.UI desing is edited

endlesslydivided committed:
1. Vocabulary choice page finished
2. Lexic category page finished
3. Change vocabulary functionality if finished. Now use cannot create same vocabularies: server logic is edited.
4. Interface enhanced and became prettier, scroller edited

endlesslydivided committed:
1. Textfield input for editing user data is created
2. Words table is created
3. Word creation form is created
4. WordToVocabulary edition and creation are enabled
5. Some errors are fixed on backend and data retrieval logic for vocabularies, wordsToVocabularies and words is edited
6. Categories, main pages are finished

endlesslydivided committed:
1. Theme changing is finished
2. Profile page is finished
3. Styles are corected

endlesslydivided committed:
Stats page is finished.
Logo is created.
Dark theme is fixed

endlesslydivided committed:
1. Client app is refactored
2. Types are written
3. Imports are sorted
